nearcore/src/config.rs

```diff
@@ -1098,32 +1099,75 @@ pub fn get_config_url(chain_id: &String) -> String {
     )
 }
 
-pub fn download_file(url: &String, path: &Path, limit: usize) {
-    actix::System::new().block_on(async move {
-        let client = awc::Client::new();
-        let mut response = client.get(url).send().await.expect("Unable to download the file");
-        // IMPORTANT: limit specifies the maximum size of the genesis or config file
-        // In case where the genesis or config file is bigger than the specified
-        // limit Overflow Error is thrown
-        let body = response
-            .body()
-            .limit(limit)
-            .await
-            .expect("File is bigger than specified limit. Please make the limit higher.");
-
-        std::fs::write(&path, &body).expect("Failed to create / write a file.");
+#[derive(thiserror::Error, Debug)]
+pub enum FileDownloadError {
+    #[error("Failed to download the file: {0}")]
+    HttpError(#[from] hyper::Error),
+    #[error("Failed to create/write to file: {0}")]
+    IoError(#[from] std::io::Error),
```

Comment:
Nit: this error variant conflates three different failure modes (open, write, move/rename). These operations fail with a same `io::Error` but have fairly distinct underlying reason. I would recommend splitting this up into an `OpenError`, `WriteError` and a `RenameError` without a `#[from]`.

Reply:
That seems like a better approach indeed. Do you happen to know if there is a less verbose way to wrap an error than:
```rust
std::fs::rename(&tmp_path, path).map_err(|e| {
    FileDownloadError::RenameError(e)
})?;
```

Reply:
Ah, as per @ChaoticTempest's comment you can directly reference the enum variant like this:
```rust
std::fs::rename(&tmp_path, path).map_err(FileDownloadError::RenameError)?;
```

Thanks for the pointer